specs/sds_detection_modules.adoc

```diff
@@ -0,0 +1,260 @@
+        # version
+        # 3. Figure out the role of the storage node (by figuring out if MON or
+        # OSD processes are running) and accordingly set the role for the node
+        # 4. Generate (or read FSID in case of ceph) and assign a temporary
+        # cluster id. Write the this detail to a defined path as well on node
+```
+
+* Similary another implementation to be introduced for gluster as well
+
+* Enhance the node agent component to load these SDS detection plugins for pre-
+defined path (tendrl/node_agent/sds_detection/plugins) to mount them
+
+* While starting the node agent service execute the SDS detection plugins and
+tag the details to the storage node
+
+* If one plugin execution succeeds stop further execution
+
+==== Sds integration impact:
+
+* No impact as such on SDS integration modules due to auto detection and their
+role comes as usual for other management activities once cluster is imported in
+Tendrl system
+
+=== Security impact:
+
+None
+
+=== Other end user impact:
+
+* While import cluster flow, the underlying SDS details would be auto-detected
+and manifested in UI for displaying layout (nodes, roles, storage system version
+etc) of the cluster beforehand
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+* API layer to take care of listing the tagged additional storage system details
+while listing the nodes for import cluster flow
+
+== Implementation:
+
+* The details of implementation are briefed in individual component impact
+sections already.
+
+* To summarize
+
+** Commons component to add a base interface for `AutoDetectPlugin`
+
+** Node agent component to extend the `AutoDetectPlugin` and define specific
+interface for SDS detection
+
+** Node agent to implement `SDSDetectPlugin` and add modules for `ceph` and
+`gluster`
+
+** Node agent to mount the SDS detection plugins while starting the service and
+execute these plugins one by one to figure out the details of the underlying
+storage system. Tag the figured out details with the storage nodes in central
+store
+
+* SDS Detect plugin for `ceph`
+
+** Execute the command `ceph version` to figure out if ceph bits are installed
+and figure out the version details well
+
+** Check the running status of mon and osd processes on the storage nodes and
+decide the role of the storage node accordingly
+
+** Get the FSID of the underlying cluster and tag the same as cluster_id. Store
+the details in pre-defined location (say /etc/tendrl/tendrl_context) as
+clsuter_id
+
+* SDS Detect plugin for `gluster`
+
+** Execute the command `gluster --version` to figure out if gluster bits are
+installed on the storage node. Also get the version details and tag with the
+node in central store
+
+** Default assign the role as `glusterd` for all the storage nodes
+
+** Execute the command `gluster pool list` and calculate as hash out of node
```

Comment:
This would be used to create a unique UUID to identify the current cluster state before import. Once cluster imported into Tendrl, it could be kept as is and no need to change the cluster_id value based on newer/fewer nodes in the cluster at later stage.

@brainfunked  comments?